File: Producto/frontend/src/remote/DTO/UserDTO.js
/**
 * DTO para representar un Usuario/Colaborador.
 * Basado en ColaboradorDTO del backend.
 */
export class UserDTO {
    constructor(data = {}) {
        this.id = data.usuarioId || data.id || null;
        this.run = data.usuarioRun || '';
        this.nombre = data.usuarioNombre || '';
        this.apellidos = data.usuarioApellidos || '';
        this.email = data.usuarioEmail || '';
        this.fechaNacimiento = data.fechaNacimiento || null;
        this.direccion = data.direccion || '';
        this.region = data.region || '';
        this.comuna = data.comuna || '';
        this.activo = data.hasOwnProperty('enabled') ? data.enabled : (data.hasOwnProperty('activo') ? data.activo : true);
        this.roles = data.roles || [];
        this.areas = data.areas || [];
    }

    static fromResponse(response) {
        if (!response || !response.data) return null;
        return new UserDTO(response.data);
    }

    static listFromResponse(response) {
        if (!response || !Array.isArray(response.data)) return [];
        return response.data.map(item => new UserDTO(item));
    }
}

File: Producto/backend/src/main/java/backend/com/gestionUsuarios/usuario/application/dto/UserDTO.java
package backend.com.gestionUsuarios.usuario.application.dto;

import backend.com.gestionUsuarios.area.application.dto.AreaDTO;
import backend.com.gestionUsuarios.role.application.dto.RoleDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDTO {

    private Long usuarioId;
    private String usuarioRun;
    private String usuarioNombre;
    private String usuarioApellidos;
    private String usuarioEmail;
    private String usuarioPassword;
    private String telefono;
    private boolean enabled;

    // Roles completos del usuario
    private Set<RoleDTO> roles;

    // Áreas completas del usuario
    private Set<AreaDTO> areas;
}
